frontend-web/webclient/app/Applications/Jobs/View.tsx

```diff
@@ -1399,37 +1487,48 @@ const RunningButtonGroup: React.FunctionComponent<{
             </Link>
         )}
-        {appType !== "WEB" || !supportsInterface ? null : (
-            <Link to={redirectToWeb ?? ""} aria-disabled={!redirectToWeb} target={"_blank"}>
-                <Button width="220px" disabled={!redirectToWeb}>
-                    <Icon name={"heroArrowTopRightOnSquare"} />
-                    <div>Open interface {hasMultipleNodes ? `(node ${rank + 1})` : null}</div>
-                </Button>
-            </Link>
-        )}
-        {appType !== "VNC" || !supportsInterface ? null : (
-            <Link to={`/applications/vnc/${job.id}/${rank}?hide-frame`} target={"_blank"} onClick={e => {
-                e.preventDefault();
-
-                const link = findDomAttributeFromAncestors(e.target, "href");
-                if (!link) return;
-
-                window.open(
-                    link,
-                    `vnc-${job.id}-${rank}`,
-                    "width=800,height=450,status=no"
-                );
-            }}>
-                <Button width="220px">
-                    <Icon name={"heroArrowTopRightOnSquare"} />
-                    <div>Open interface {hasMultipleNodes ? `(node ${rank + 1})` : null}</div>
-                </Button>
-            </Link>
+        {interfaceLinks.length < 1 ? <></> : (
+            <Flex>
+                <Link to={interfaceLinks[0].link ?? ""} aria-disabled={!interfaceLinks[0]} target={"_blank"}>
+                    <Button attachedLeft={interfaceLinks.length > 1} disabled={!interfaceLinks[0]}>
+                        <Icon name="heroArrowTopRightOnSquare" />
+                        <div style={{minWidth: interfaceLinks.length > 1 ? "130px" : "164px"}}>{interfaceLinks[0].target ?? ("Open interface" + (hasMultipleNodes ? ` (node ${rank + 1})` : null))}</div>
+                    </Button>
+                </Link>
+                <ClickableDropdown
+                    trigger={
+                        <div className={InterfaceSelectorTrigger}> 
+                            <Icon name="chevronDownLight"/>
+                        </div>
+                    }
+                    right={"0px"}
+                    minWidth={"245px"}
+                    paddingControlledByContent
+                >
+                    {interfaceLinks.slice(1).map((link, k) => (
+                        supportsInterface ? 
+                            <Link
+                                to={link.link ?? ""}
+                                key={link.link ?? (link.rank + (link.port?.toString() ?? ""))}
+                            >
+                                <Flex
+                                    key={k}
+                                    gap="5px"
+                                    alignItems={"center"}
+                                    p={8}
+                                >
+                                    <Icon name="heroArrowTopRightOnSquare" />
+                                    {link.target ?? "Open interface"}
```

Comment:
Depending on the length of the `link.target` value, it might make sense to wrap it in a `Truncate`-tag.

Reply:
Jonas has a valid point given that `link.target` is coming from a user-specified value. It might be very long.